proposals/NNNN-Deprecated-interfaces-markup.md

```diff
@@ -0,0 +1,133 @@
+# Mark public deprecated methods
+
+* Proposal: [SDL NNN](NNNN-NNNN-Deprecated-interfaces-markup.md)
+* Author: Alexander Kutsan
+* Status: **Awaiting review**
+* Impacted Platforms: Core
+
+## Introduction
+Public methods is part of SDL interface. New features may use some existing interfaces and expect some behavior. 
+Sometimes some features use interfaces that planed to be removed. And after removing interfaces feature can't be released due to conflicts in code.
+
+## Motivation
+Marking method as deprecated should trigger a compile time warning, but should not block project assemble.
+Reverse feature support plugins that can be created by 3rd party developers.
+Each public method should work to handle backward capabilities with other developers plugins code.
+Also some classes also can be deprecated and creating instances of this classes should cause compile time warnings also.
+
+## Proposed Solution
+Deprecating methods should enough explicit in code and in documentation. 
+Each deprecated method should be marked such in documentation page and if this method is used in code, it should trigger compile time warning.
+SDL use doxygen for documentation public interfaces, and doxygen support [@deprecated](https://www.stack.nl/~dimitri/doxygen/manual/commands.html#cmddeprecated) command.
+In code deprecated metods should be done with macro that will use c++14 [ [[deprecated]] ] (http://en.cppreference.com/w/cpp/language/attributes)
+atribute or other featured depends from used compiler and c++ version. 
```

Comment:
Please reword this sentence.  I believe it should be "Attributes or other features depend on used compiler and c++ version."